static/app/views/settings/account/accountSecurity/accountSecurityDetails.tsx

```diff
@@ -43,160 +49,165 @@ function AuthenticatorDate({label, date}: AuthenticatorDateProps) {
+    if (!authenticator?.authId) {
+      return;
     }
+    rename({id: authenticator.authId, device, name: deviceName});
   };
 
-  renderBody() {
-    const {authenticator} = this.state;
-
-    if (!authenticator) {
-      return null;
-    }
+  if (isAuthenticatorPending || isRemoveLoading || isRenameLoading) {
+    return <LoadingIndicator />;
+  }
 
-    const {deleteDisabled, onRegenerateBackupCodes} = this.props;
-
-    return (
-      <Fragment>
-        <SettingsPageHeader
-          title={
-            <Fragment>
-              <span>{authenticator.name}</span>
-              <AuthenticatorStatus
-                data-test-id={`auth-status-${
-                  authenticator.isEnrolled ? 'enabled' : 'disabled'
-                }`}
-                enabled={authenticator.isEnrolled}
-              />
-            </Fragment>
-          }
-          action={
-            <AuthenticatorActions>
-              {authenticator.isEnrolled && authenticator.allowRotationInPlace && (
-                <LinkButton
-                  to={`/settings/account/security/mfa/${authenticator.id}/enroll/`}
-                >
-                  {t('Rotate Secret Key')}
-                </LinkButton>
-              )}
-              {authenticator.isEnrolled && authenticator.removeButton && (
-                <Tooltip
-                  title={t(
-                    "Two-factor authentication is required for at least one organization you're a member of."
-                  )}
-                  disabled={!deleteDisabled}
-                >
-                  <RemoveConfirm onConfirm={this.handleRemove} disabled={deleteDisabled}>
-                    <Button priority="danger">{authenticator.removeButton}</Button>
-                  </RemoveConfirm>
-                </Tooltip>
-              )}
-            </AuthenticatorActions>
-          }
-        />
-
-        <TextBlock>{authenticator.description}</TextBlock>
-
-        <AuthenticatorDates>
-          <AuthenticatorDate label={t('Created at')} date={authenticator.createdAt} />
-          <AuthenticatorDate label={t('Last used')} date={authenticator.lastUsedAt} />
-        </AuthenticatorDates>
-
-        <U2fEnrolledDetails
-          isEnrolled={authenticator.isEnrolled}
-          id={authenticator.id}
-          devices={authenticator.devices}
-          onRemoveU2fDevice={this.handleRemove}
-          onRenameU2fDevice={this.handleRename}
-        />
-
-        {authenticator.isEnrolled && authenticator.phone && (
-          <PhoneWrapper>
-            {t('Confirmation codes are sent to the following phone number')}:
-            <Phone>{authenticator.phone}</Phone>
-          </PhoneWrapper>
-        )}
-
-        <RecoveryCodes
-          onRegenerateBackupCodes={onRegenerateBackupCodes}
-          isEnrolled={authenticator.isEnrolled}
-          codes={authenticator.codes}
-        />
-      </Fragment>
-    );
+  if (!authenticator || isError) {
```

Comment:
shouldn't we check only for `isError` here?

Reply:
and if an error happens in the mutations?

Reply:
The previous code handled the response as if the endpoint could return `null`.
As far as I see from the python code that should not be possible, will adjust the types.